sycl/test-e2e/InvokeSimd/Spec/simd_mask_merge.cpp

```diff
@@ -0,0 +1,117 @@
+// TODO: enable on Windows once driver is ready
+// REQUIRES: gpu && linux
+// UNSUPPORTED: cuda || hip
+//
+// Check that full compilation works:
+// RUN: %clangxx -fsycl -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr %s -o %t.out
+// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+//
+// VISALTO enable run
+// RUN: env IGC_VISALTO=63 IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %GPU_RUN_PLACEHOLDER %t.out
+
+
+#include <sycl/detail/boost/mp11.hpp>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/ext/oneapi/experimental/invoke_simd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <functional>
+#include <iostream>
+#include <type_traits>
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental;
+namespace esimd = sycl::ext::intel::esimd;
+constexpr int VL = 16;
+
+[[intel::device_indirectly_callable]]
+simd<float, VL>
+SIMD_CALLEE(simd<float, VL> va,
+             simd_mask<float, VL> mask) SYCL_ESIMD_FUNCTION {
+  esimd::simd<float, VL> ret;
+  esimd::simd_mask<VL> emask;
+  for(int i = 0; i < VL; i++)
+    emask[i] = static_cast<bool>(mask[i]);
+  ret.merge(va, !emask);
+  return ret;
+}
+
+int main() {
+  sycl::queue q{gpu_selector_v};
+  auto dev = q.get_device();
+
+  std::cout << "Running on " << dev.get_info<sycl::info::device::name>()
+            << "\n";
+ constexpr unsigned Size = 1024;
+  constexpr unsigned GroupSize = 4 * VL;
+
+  auto ctxt = q.get_context();
+
+  float *A =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+  float *C =
+      static_cast<float *>(malloc_shared(Size * sizeof(float), dev, ctxt));
+
+  bool *M = static_cast<bool *>(malloc_shared(Size * sizeof(bool), dev, ctxt));
```

Comment:
Any particular reason for USM over buffers? I think `malloc_shared` is an optional feature and requires us to check for its `sycl::aspect`.

And even if we are to use USM, there are overloads of these APIs that accept `sycl::queue` and a templated type, something like

```
  auto *M = malloc_shared<bool>(Size, q);
```

Reply:
sorry i just copied simd_mask.cpp, i'll fix this